gui/tools/Helper.h

```diff
@@ -0,0 +1,227 @@
+#include <vector>
+#include <sstream>
+#include <string>
+
+// VTK includes
+#include <vtksys/SystemTools.hxx>
+#include "vtkMatrix3x3.h"
+#include "vtkMatrix4x4.h"
+#include "vtkNew.h"
+#include "vtkTransform.h"
+
+namespace help
+{
+  //----------------------------------------------------------------------------
+  // Description
+  // Split string by separated char
+  static void SplitString(const std::string &s, char delim,
+                          std::vector<std::string> &elems)
+  {
+    std::stringstream ss(s);
+    std::string item;
+    while (std::getline(ss, item, delim))
+      {
+      elems.push_back(item);
+      }
+  }
+
+  //----------------------------------------------------------------------------
+  // Description
+  // Get the filename path (copy from vtk in order to allow debug)
+  static std::string GetFilenamePath(const std::string& filename)
+  {
+    std::string fn = filename;
+    vtksys::SystemTools::ConvertToUnixSlashes(fn);
+
+    std::string::size_type slash_pos = fn.rfind("/");
+    if (slash_pos != std::string::npos)
+    {
+      std::string  ret = fn.substr(0, slash_pos);
+      if (ret.size() == 2 && ret[1] == ':')
+      {
+        return ret + '/';
+      }
+      if (ret.empty())
+      {
+        return "/";
+      }
+      return ret;
+    }
+    else
+    {
+      return "";
+    }
+  }
+
+  //----------------------------------------------------------------------------
+  // Description
+  // Read global path and extract all contained path
+  static std::vector<std::string> ExtractAllFilePath(const char* globalPath)
+  {
+    std::vector<std::string> pathList;
+
+    // Open file which contains the list of all file
+    std::ifstream container(globalPath);
+    if (!container.is_open())
+    {
+      std::cerr << "Unable to open : " << globalPath << std::endl;
+      return pathList;
+    }
+
+    // Extract path of globalPath from globalPath
+    //std::string pwd_str = globalPath;
+    std::string directoryPath = help::GetFilenamePath(std::string(globalPath));
+    // Get current working directory
+    if (directoryPath == "")
+    {
+      directoryPath = vtksys::SystemTools::GetCurrentWorkingDirectory();
+    }
+
+    std::string path;
+    while (!container.eof())
+    {
+      std::getline(container, path);
+      // only get the file name, not the whole path
+      std::vector <std::string> elems;
+      help::SplitString(path, ' ', elems);
+
+      // check if there are an empty line
+      if (elems.size() == 0)
+      {
+        continue;
+      }
+
+      // Create the real data path to access depth map file
+      pathList.push_back(directoryPath + "/" + elems[elems.size() - 1]);
+    }
+
+    return pathList;
+  }
+
+  //----------------------------------------------------------------------------
+  // Description
+  // Read global path and extract all contained path
+  static std::vector<std::string> ExtractAllKRTDFilePath(const char* globalPath, const char* framelist)
```

Comment:
I got a build warning that this function is unused.  That's good!  Let's remove it.